src/main/java/com/squareup/javapoet/TypeSpec.java

```diff
@@ -317,32 +316,44 @@ public Builder addAnnotation(AnnotationSpec annotationSpec) {
       return this;
     }
 
-    public Builder addAnnotation(Type annotation) {
+    public Builder addAnnotation(TypeName annotation) {
       return addAnnotation(AnnotationSpec.builder(annotation).build());
     }
 
+    public Builder addAnnotation(Type annotation) {
```

Comment:
I think we want `Class<?>` / `ClassName` for annotations. I've changed 'em all over to that.